bin/schema_generator.sh

```diff
@@ -70,18 +70,21 @@ function clone_and_configure_mps() {
 
 function prepare_metadata() {
     local telemetry_metadata="metadata/telemetry-ingestion/telemetry-ingestion.1.schema.json"
+    local pioneer_metadata="metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json"
     local structured_metadata="metadata/structured-ingestion/structured-ingestion.1.schema.json"
 
-    find ./telemetry -name "*.schema.json" -type f \
+    find ./telemetry \
+        -name "*.schema.json" -type f \
```

Comment:
We should consider recasting this logic now as a single loop through schema files with an if/then chain:

```
for schema in $(find . -name "*.schema.json" -type f); do
  if [[ "$schema" ~= "telemetry/.*" ]]; then
    metadata_merge $telemetry_metadata $schema
  elif ...
done
```

That would prevent the need to ensure we match the inverse logic for the structured case.

Reply:
Apparently relying on the output of find is fragile due the possibility of whitespace and globs in the output (https://github.com/koalaman/shellcheck/wiki/SC2044). The structure of the schema repository is enforced at compile time, so it should be fine.